scripts/patchEsbuildDist.ts

```diff
@@ -0,0 +1,27 @@
+// esbuild 0.14.4 https://github.com/evanw/esbuild/blob/master/CHANGELOG.md#0144 introduced a
+// change that breaks the "overwrite for cjs require('...')() usage" hack used in plugin-vue
+// and plugin-react. For the moment, we can remove the extra exports code added in 0.14.4 to
+// continue using it.
+
+import { bold } from 'picocolors'
+import { readFileSync, writeFileSync } from 'fs'
+
+const indexPath = process.argv[2]
+const varName = process.argv[3]
+
+let code = readFileSync(indexPath, 'utf-8')
+
+// overwrite for cjs require('...')() usage
+code = code.replace(
+  `module.exports = __toCommonJS(src_exports);`,
```

Comment:
is this the only use of __toCommonJS and src_exports? if so should they be removed too?

This feels brittle and targets an implementation detail of esbuild. 
at the very least there should be safeGuards validating that varName and the string about to be replaced exist.

Reply:
+1, if we end up doing this, we should add the safeGuards. The tests are going to tell us if these lines aren't inserted though but it is better if we find the error here instead of on CI.

For reference, I wanted to test the approach that @antfu suggested and push this PR in case it is useful for others looking into this problem. What we get at the end is the same issue as in https://github.com/vitejs/vite/pull/6329. It may be better to go with #6329 here and fix the bug. Both rollup and esbuild >0.14.4 agree that there is a problem on our side, that esbuild 0.14.3 is hiding.